File: leetcode_problems/Containerwithmostwater.py
height = [1,8,6,2,5,4,8,3,7]
res=0
for l in range(len(height)):
    for r in range(l+1,len(height)):
        area = (r-l)*min(height[l],height[r])
        res=max(res,area)
print(res)

      
  

File: DS_and_A/neetcodeBAsics/containerwithmostwater.java
package neetcodeBAsics;

public class containerwithmostwater {

    private static int sol1(int[]arr){
        int l =0;
        int r=arr.length-1;
        int currentlength=arr.length-1;
        int maxcapacity=0;
        while (l<r) {
            maxcapacity=Math.max(maxcapacity,currentlength*(Math.min(arr[l], arr[r])));
            if(arr[l]<arr[r]){
                l++;
            }else{
                r--;
            }
            currentlength--;
        }
       return maxcapacity;
    }
    public static void main(String[] args) {
        int[]  height = {1,1};
        sol1(height);
    }
}
